swap_meet/vendor.py

```diff
@@ -1,2 +1,45 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+            
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, friend, my_item, their_item):  
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        friend.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        friend.inventory.append(my_item)
+        return True
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory:
+            return False
+
+        return self.swap_items(friend, self.inventory[0], friend.inventory[0])
+
+    def get_best_by_category(self, category):
+        return max(self.get_by_category(category), key = lambda item: item.condition, default = None)
```

Comment:
Would a guard clause make more sense before returning max? For example:

`if not self.inventory or not other.inventory:
            return False
return max(self.get_by_category(category), key = lambda item: item.condition)`

Or is this better:
`return max(self.get_by_category(category), key = lambda item: item.condition, default = None)`

Reply:
In general, I would prefer using `default=` since it's built directly into how `max` operates. `max` doesn't know what a reasonable default value for all cases since it doesn't know what data it might be given. But we know how we're calling it, and can tell it what that reasonable default is for our situation.